### PR TITLE
feat: update provider api

### PR DESCRIPTION
**Pull request by yqrashawn**
PORTAL-2298
PORTAL-2378

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/helios/742)

### CODE REVIEW

**Comment by angelia-yuqi-personal** [comment]
Task linked: [PORTAL-2298 支持多链](https://app.clickup.com/t/3719212/PORTAL-2298)

**Comment by angelia-yuqi-personal** [comment]
Task linked: [PORTAL-2378 去掉兼容 Portal 的功能，增加一个开关](https://app.clickup.com/t/3719212/PORTAL-2378)

File: packages/inpage/setup-inpage-provider.js [hunk 1]
```diff
@@ -12,31 +12,78 @@ function validateMessage(e) {
...
+    )
+      return
+    window.localStorage.setItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX, false)
+    window.location.reload()
+  }
+}
+
+function maybeOverrideWindowDotEthereum(overrideWindowDotEthereum) {
+  if (overrideWindowDotEthereum) {
+    if (window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM)) {
```
Comment (angelia-yuqi-personal) [inline packages/inpage/setup-inpage-provider.js:61]: when window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM), should return

**Review by angelia-yuqi-personal — CHANGES_REQUESTED** [review]

File: packages/inpage/setup-inpage-provider.js (hunk 1 above)
Comment (yqrashawn) [reply packages/inpage/setup-inpage-provider.js]: fixed with new push

**Review by angelia-yuqi-personal — APPROVED** [review]

**Pull request merged**